randovania/generator/old_generator_reach.py

```diff
@@ -231,7 +231,8 @@ def connected_nodes(self) -> Iterator[Node]:
         :return:
         """
         self._calculate_reachable_paths()
-        all_nodes = self.all_nodes
+        assert self._reachable_paths is not None
+        all_nodes = typing.cast(tuple[Node, ...], self.all_nodes)
```

Comment:
what's with all the casting for this? is that really desirable?

Reply:
it's a weird thing. The `all_nodes` array does in fact have holes in it, if you iterate.
But the way we do it here, we're always using indices that are guaranteed to not be None.

Reply:
huh, ok. would it be worth making a helper so that we don't need to manually cast this every time?

Reply:
Maybe? Annoyingly this is all in very hot code.

Reply:
ugh, right. so the overhead actually matters. I guess this is fine then